lib/src/client.dart

```diff
@@ -521,13 +536,40 @@ class Client extends MatrixApi {
             loginTypes.map((f) => f.type ?? '').toSet(), supportedLoginTypes);
       }
 
-      return (wellKnown, versions, loginTypes);
+      return (versions, loginTypes);
     } catch (_) {
       homeserver = null;
       rethrow;
     }
   }
 
+  /// Gets discovery information about the domain. The file may include
+  /// additional keys, which MUST follow the Java package naming convention,
+  /// e.g. `com.example.myapp.property`. This ensures property names are
+  /// suitably namespaced for each application and reduces the risk of
+  /// clashes.
+  ///
+  /// Note that this endpoint is not necessarily handled by the homeserver,
+  /// but by another webserver, to be used for discovering the homeserver URL.
+  ///
+  /// In case is provided, [hostname] will be used in place of [homeserver] for
+  /// the HTTPS lookup. This is useful since the matrix homeserver does not
+  /// necessarily also host a copy of the `.well-known`.
+  ///
+  /// The result of this call is stored in [wellKnown] for later use at runtime.
+  @override
+  Future<DiscoveryInformation> getWellknown([String? hostname]) async {
```

Comment:
This doesn't add any value for us now. I would prefer that this adds a cache-lifetime and returns the cached version if the lifetime has not expired, or fetches again from the server and stores it then in the db

Reply:
I'm going to merge this as is for now, we can give this function a cache time at a later point. Users usually don't need to call this anyway, since they can access the well-known synchronously using the property.